File: app/src/main/java/mn/tqt/dummy/DummyController.java
package mn.tqt.dummy;

import org.apache.kafka.clients.producer.KafkaProducer;
import org.apache.kafka.clients.producer.ProducerConfig;
import org.apache.kafka.clients.producer.ProducerRecord;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;
import java.util.Properties;

@RestController
public class DummyController {

    @GetMapping("/generate-dummy-data")
    public void setupDummyTopic() throws InterruptedException {
        var props = new Properties();

        props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9094");
        props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG,
                "org.apache.kafka.common.serialization.IntegerSerializer");
        props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
                "io.confluent.kafka.serializers.KafkaJsonSerializer");
        props.put("schema.registry.url", "localhost:9091");

        var producer = new KafkaProducer<Integer, DummyObject>(props);

        var nestedObject = new mn.tqt.dummy.NestedObject(2, List.of(
                new mn.tqt.dummy.Entry(1, Map.of(1, "a", 2, "b")),
                new mn.tqt.dummy.Entry(2, Map.of(3, "c", 4, "d", 5, "e"))));
        for (var i = 0; i < 100; i++) {
            var data = new mn.tqt.dummy.DummyObject(i, nestedObject);
            var record = new ProducerRecord<>("my-topic", data.id(), data);
            Thread.sleep(2);
            producer.send(record);
        }

        producer.close();

    }
}

File: localKafkaInitializer/src/main/kotlin/mn/tqt/local_kafka_initializer/DummyController.kt
package mn.tqt.local_kafka_initializer

import org.apache.kafka.clients.producer.KafkaProducer
import org.apache.kafka.clients.producer.ProducerConfig
import org.apache.kafka.clients.producer.ProducerRecord

class DummyController {

    fun populateKafka() {
        val props = java.util.Properties()
        props[ProducerConfig.BOOTSTRAP_SERVERS_CONFIG] = "localhost:9094"
        props[ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG] =
            "org.apache.kafka.common.serialization.StringSerializer"
        props[ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG] =
            "io.confluent.kafka.serializers.KafkaJsonSerializer"
        props["schema.registry.url"] = "localhost:8081"

        val producer = KafkaProducer<String, SomeObject>(props)

        val nestedObject = NestedObject(listOf(Entry(1.5), Entry(2.0)))
        for (i in 1..100) {
            val data = SomeObject(java.util.UUID.randomUUID(), listOf("a", "b"), nestedObject)
            val record = ProducerRecord("my-topic", data.id.toString(), data)
            Thread.sleep(2)
            producer.send(record)
        }


    }
}

data class SomeObject(
    val id: java.util.UUID, val strings: List<String>, val nestedObject:
    NestedObject
)

data class NestedObject(val entries: List<Entry>)

data class Entry(val entryVal1: Double?)